api/cpp/esp-idf/slint/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+# Copyright © SixtyFPS GmbH <info@slint.dev>
+# SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+idf_component_register(
+  SRCS "src/esp_slint.cpp"
+  INCLUDE_DIRS "include"
+  REQUIRES "esp_lcd" "esp_lcd_touch"
+  PRIV_REQUIRES "esp_timer")
+
+set(Rust_CARGO_TARGET "xtensa-${IDF_TARGET}-none-elf")
```

Comment:
Would it b possible here to use `${IDF_TARGET_ARCH}` instead of hard-coding xtensa?

Reply:
Not really.  Because the riscv target use different naming conventions

Reply:
Ahh right. Maybe it would make sense here then to check `${IDF_TARGET_ARCH}` and if it's _not_ `xtensa`, produce a cmake error - instead of going forward and producing much funnier messages later (maybe at link time?).